### PR TITLE
refactor: reworks kubo-rpc as an openapi server

### PR DESCRIPTION
**Pull request by nathanielc**
With this change we no longer use actix-web for the kubo-rpc server. Instead we use openapi and hyper. Additionally we run the both APIs under the same TCP port.

Also all uses of unimock have been replaced with mockall as it has proven to be much easier to use and assert behavior.

### CODE REVIEW

File: api/ceramic.yaml [hunk 1]
```diff
@@ -11,8 +11,11 @@ info:
     name: MIT
     url: https://mit-license.org/
 
+servers:
+  - url: /ceramic
+
 paths:
-  '/ceramic/subscribe/{sort_key}/{sort_value}':
+  '/subscribe/{sort_key}/{sort_value}':
```
[the remark this answers is not in the document]
Comment (AaronGoldman) [reply api/ceramic.yaml:18]: I would default to not having a version unless we need it.
It is just one more thing to deal with and we can use a new name if we want a new behavior

File: kubo-rpc-server/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+target
```
Comment (AaronGoldman) [inline kubo-rpc-server/.gitignore:1]: why do we need this rather than just the top level

**Review by AaronGoldman — APPROVED** [review]
This is long but I think it looks good.
That patch may be fragile to changes in the generated code.
How do you think we would detect it if that broke?

**Comment by nathanielc** [comment]
> How do you think we would detect it if that broke?

Applying the patch is a CI step so if the patch fails to apply CI will fail. Additionally we have locked down the version of the code generator we use so the generated code should not change unexpectedly. So I imagine a workflow where we update the code generator and the patch in the same PR and both need to be working before CI will let it merge.

File: kubo-rpc-server/.gitignore (hunk 2 above)
Comment (nathanielc) [reply kubo-rpc-server/.gitignore:1]: We don't however the generated code doesn't know we are in a workspace so it adds this file. Thankfully is a noop.

**Review by dbcfd — APPROVED** [review]

**Review by smrz2001 — APPROVED** [review]
🚀

**Pull request merged**